p2p/simulations/simulation.go

```diff
@@ -49,9 +49,11 @@ func (s *Simulation) Run(ctx context.Context, step *Step) (result *StepResult) {
 	defer stop()
 
 	// perform the action
-	if err := step.Action(ctx); err != nil {
-		result.Error = err
-		return
+	if step.Action != nil {
```

Comment:
Why would `step.Action` be `nil`? Should it be possible?

Reply:
i've seen places around the codebase where empty actions that don't do anything and return `nil` are passed to the simulation. i.e. discovery simulation:
https://github.com/ethersphere/go-ethereum/blob/9e9fc87e70accf2b81be8772ab2ab0c914e95666/swarm/network/simulations/discovery/discovery_test.go#L247-L249

Reply:
I don't like this. Can we please remove it. Or fix it in simulations. Action should never be `nil`

Reply:
Reverted from master.